azext_iot/iothub/_help.py

```diff
@@ -310,6 +310,323 @@ def load_iothub_help():
         short-summary: Upload a local file as a device to a pre-configured blob storage container.
     """
 
+    helps[
+        "iot hub message-endpoint"
+    ] = """
+        type: group
+        short-summary: Manage custom endpoints of an IoT hub.
+    """
+
+    helps[
+        "iot hub message-endpoint create"
+    ] = """
+        type: group
+        short-summary: Add an endpoint to an IoT Hub.
+    """
+
+    helps[
+        "iot hub message-endpoint create cosmosdb-container"
+    ] = """
+        type: command
+        short-summary: Add a Cosmos DB Container endpoint for an IoT Hub.
+        examples:
+          - name: Create a key-based Cosmos DB Container endpoint for an IoT Hub.
+            text: >
+                az iot hub message-endpoint create cosmosdb-container -n {iothub_name} --en {endpoint_name} --container {container}
+                --db {database} --endpoint-account {account_name}
+          - name: Create a Cosmos DB Container endpoint for an IoT Hub using a connection string.
+            text: >
+                az iot hub message-endpoint create cosmosdb-container -n {iothub_name} --en {endpoint_name} -c {connection_string}
+                --container {container} --db {database}
+          - name: Create a Cosmos DB Container endpoint for an IoT Hub using the specified primary key and endpoint uri.
+            text: >
+                az iot hub message-endpoint create cosmosdb-container -n {iothub_name} --en {endpoint_name} --pk {primary_key}
+                --endpoint-uri {endpoint_uri} --container {container} --db {database}
+          - name: Create a Cosmos DB Container endpoint for an IoT Hub using system assigned identity and a partition key name.
+                  The partition key template will be the default.
+            text: >
+                az iot hub message-endpoint create cosmosdb-container -n {iothub_name} --en {endpoint_name} --endpoint-uri {endpoint_uri}
+                --container {container} --db {database} --auth-type identityBased --pkn {partition_key_name}
+          - name: Create a Cosmos DB Container endpoint for an IoT Hub using user assigned identity, partition key name, and partition key template.
+            text: >
+                az iot hub message-endpoint create cosmosdb-container -n {iothub_name} --en {endpoint_name} --endpoint-uri {endpoint_uri}
+                --container {container} --db {database} --auth-type identityBased --pkn {partition_key_name} --pkt {partition_key_template}
```

Comment:
We should eliminate/deprecate `--auth-type` in favor of determining auth type implicitly based on provided identity params (or lack of them) --- similar to ADT UAI changes.